File: lua/002/main.lua
--[[
    ProjectEuler 002
    Lua / main.lua
    copyright (c) 2015 Susisu

    Lua 5.1.5
    `lua main.lua`
]]

function main()
    local a = 0
    local b = 1
    local tot = 0
    while b < 4000000 do
        if b % 2 == 0 then
            tot = tot + b
        end
        local t = a
        a = b
        b = b + t
    end
    print(tot)
end

main()

File: rust/002/main.rs
/*
    ProjectEuler 002
    Rust / main.rs
    copyright (c) 2015 Susisu

    Rust 1.3.0
    `rustc main.rs -o bin/main`
*/

fn main() {
    let n = calc(0, 1, 2, 4000000);
    println!("{}", n);
}

fn calc(acc: i32, a: i32, b: i32, max: i32) -> i32 {
    if a <= max {
        calc(if a % 2 == 0 { acc + a } else { acc }, b, a + b, max)
    }
    else {
        acc
    }
}
